docs/RFCS/20220608_split_jobs_system_tables.md

```diff
@@ -0,0 +1,479 @@
+relies on scanning the jobs table to show what jobs are currently running, to find jobs which need
+to start running, and to find those which are running but need to be paused or cancelled. Further,
+it is critically important that an operator facing a production incident be able to inspect what
+jobs are running and pause or cancel one if it is suspected of being the cause of problems. However
+today, those actions can be blocked by any job that has a slow or frequent transaction that has
+locked one or more rows in the jobs table. 
+
+The jobs system and its associated operator-facing tools should be resilient to the behaviour of any
+individual job, but it is not today, and this has caused and continues to cause numerous production
+issues.
+
+To avoid such contention related issues, as well as other issues such as excessive MVCC garbage,
+many jobs end up electing to store less state or updating it less often than they would like. Such
+jobs may lose progress if paused or restarted, or are forced to hold more state in memory instead,
+adding scale limits or risking OOMs. 
+
+In addition to contention issues, if a job adds to much information to its current state record, or
+updates it too often, the MVCC garbage associated with those revisions to its row in the jobs table
+can cause the table to become unavailable because CockroachDB's KV layer is not permitted to split
+between revisions of a single SQL row. Packing state, progress, and control all into one row leads
+to larger rows, increasing the risk of this type of failure. While this is partially mitigated today
+via careful organisation into column families, the large serialised protobuf messages favoured by
+jobs to represent progress or arguments still often result in large job rows, which are typically
+edited every 30-60sec to persist progress. Jobs should be able to store all the state that is useful
+to them without running into limitations of the job system. They should not need to discard useful
+state for fear of compromising the job control system itself, and the associated APIs offered to
+jobs should make it easier to write jobs that do what their authors want to do without worrying
+about adversely affecting the jobs system.
+
+## Goals
+
+1. The jobs system should allow operators to inspect and control the set of jobs, regardless of how
+   transactions are used to create jobs or update the persisted state of any one job.
+2. Jobs should be able to be created in a transaction, with typical transactional semantics, i.e. a
+   job is only created and starts running if the transaction commits and minimal effect if it rolls
+   back.
+3. Jobs should be able to store as much persisted state as they want, and update it as much as they
+   want, without negatively affecting other jobs or the jobs system health.
+4. Operators should be able to inspect the recent history of a job's status or progress.
+
+## Design Summary
```

Comment:
Since in several of these tables we insert a new row whenever there's an update. Do we want to make a periodic cleanup of older jobs data (like older than one year)?

Reply:
jobs today are already deleted 14 days after moving to a terminal state; that deletion would now mean deleting from each of these tables by job ID vs just the one table today, but conceptually still the same behavior which is that the job is gone after 14d.